cpp/dolfin/fem/utils.cpp

```diff
@@ -137,14 +137,14 @@ la::PETScMatrix dolfin::fem::create_matrix(const Form& a)
 
   // Create and build sparsity pattern
   la::SparsityPattern pattern(mesh.mpi_comm(), index_maps);
+  SparsityPatternBuilder::get_cell_dofs_function get_cell_dofs = 
+    [&dofmaps](unsigned int i, int cell_index) { return dofmaps[i]->cell_dofs(cell_index); };
```

Comment:
Will any resonable compiler be able to avoid creating a copy of the temporary returned by `dofmaps[i]->cell_dofs(cell_index)`?

Reply:
May be use https://en.cppreference.com/w/cpp/utility/functional/mem_fn?

I would like ultimately to get just pass a single array of dofs (maybe with on offset array to handle a variable numbers) and pass the single array.

Reply:
Not sure how to use mem_fn. Since it seems to be designed to only work with class members, I guess I would put it around the inner call to `dofmaps[i]->cell_dofs(cell_index)`, but the (possible) copy is happening in the outer layer.

Reply:
The links in the first answer at https://stackoverflow.com/questions/41118520/when-is-return-by-value-ok might be relevant to my original question.